crates/red_knot_module_resolver/src/resolver.rs

```diff
@@ -465,35 +415,119 @@ mod tests {
             resolve_module(&db, functools_module_name).as_ref()
         );
 
-        assert_eq!(&stdlib_dir, functools_module.search_path().path());
+        assert_eq!(stdlib_dir, functools_module.search_path().to_path_buf());
         assert_eq!(ModuleKind::Module, functools_module.kind());
-        assert_eq!(&functools_path.clone(), functools_module.file().path(&db));
+
+        let expected_functools_path =
+            VfsPath::FileSystem(custom_typeshed.join("stdlib/functools.pyi"));
+
+        assert_eq!(&expected_functools_path, functools_module.file().path(&db));
 
         assert_eq!(
             Some(functools_module),
-            path_to_module(&db, &VfsPath::FileSystem(functools_path))
+            path_to_module(&db, &expected_functools_path)
         );
 
         Ok(())
     }
 
+    fn create_module_names(raw_names: &[&str]) -> Vec<ModuleName> {
+        raw_names
+            .iter()
+            .map(|raw| ModuleName::new(raw).unwrap())
+            .collect()
+    }
+
     #[test]
-    fn first_party_precedence_over_stdlib() -> anyhow::Result<()> {
+    fn stdlib_resolution_respects_versions_file_py38() {
         let TestCase {
             db,
-            src,
             custom_typeshed,
             ..
-        } = create_resolver()?;
+        } = create_resolver_builder().unwrap().build();
+
+        let existing_modules = create_module_names(&["asyncio", "functools", "xml.etree"]);
+        for module_name in existing_modules {
+            let resolved_module = resolve_module(&db, module_name.clone()).unwrap_or_else(|| {
+                panic!("Expected module {module_name} to exist in the mock stdlib")
+            });
+            let search_path = resolved_module.search_path();
+            assert_eq!(
+                *custom_typeshed.join("stdlib"),
+                search_path,
+                "Search path for {module_name} was unexpectedly {search_path:?}"
+            );
+            assert!(
+                search_path.is_stdlib_search_path(),
+                "Expected a stdlib search path, but got {search_path:?}"
+            );
+        }
 
-        let stdlib_dir = custom_typeshed.join(TYPESHED_STDLIB_DIRECTORY);
-        let stdlib_functools_path = stdlib_dir.join("functools.py");
-        let first_party_functools_path = src.join("functools.py");
+        let nonexisting_modules = create_module_names(&[
+            "collections",
+            "importlib",
+            "importlib.abc",
+            "xml",
+            "asyncio.tasks",
+        ]);
+        for module_name in nonexisting_modules {
+            assert!(
+                resolve_module(&db, module_name.clone()).is_none(),
+                "Unexpectedly resolved a module for {module_name}"
+            );
+        }
```

Comment:
I think I would move this into its own test. The boilerplate is fairly limited and it is testing something else.